arch/arm/core/thread.c

```diff
@@ -148,6 +152,15 @@ void z_new_thread(struct k_thread *thread, k_thread_stack_t *stack,
 FUNC_NORETURN void z_arch_user_mode_enter(k_thread_entry_t user_entry,
 	void *p1, void *p2, void *p3)
 {
+#if defined(CONFIG_USERSPACE) && \
+	defined(CONFIG_MPU_REQUIRES_POWER_OF_TWO_ALIGNMENT)
+	/* Guard was carved-out of the stack object, not reserved in advance.
+	 * Now that we are entering user mode, the guard is moved to the
+	 * privilege stack, un-do the space carved out.
+	 */
+	_current->stack_info.start -= MPU_GUARD_ALIGN_AND_SIZE;
```

Comment:
That's not really needed, because of the way we configure the thread stack access :)
```
	/* Thread user stack */
	LOG_DBG("configure user thread %p's context", thread);
	if (thread->arch.priv_stack_start) {
		u32_t base = (u32_t)thread->stack_obj;
		u32_t size = thread->stack_info.size +
			(thread->stack_info.start - base);

``` 

So this will waste a bit of run-time performance and flash size.

Reply:
That should be changed to simply use the bounds in thread->stack_info.
It's intended that thread->stack_info reflects the actual user accessible buffer.

Reply:
addressed